src/frontend/komponenter/Fagsak/Opprett/OpprettBehandling.tsx

```diff
@@ -1,94 +1,46 @@
-import { Knapp } from 'nav-frontend-knapper';
-import { Systemtittel } from 'nav-frontend-typografi';
 import * as React from 'react';
-import { useHistory } from 'react-router';
-import { axiosRequest } from '../../../api/axios';
-import { IFagsak } from '../../../typer/fagsak';
-import { Valideringsstatus } from '../../../typer/felt';
-import { Ressurs, RessursStatus } from '../../../typer/ressurs';
-import { actions as fagsakActions, useFagsakDispatch } from '../../FagsakProvider';
-import {
-    actions,
-    useOpprettBehandlingContext,
-    useOpprettBehandlingDispatch,
-} from './OpprettBehandlingProvider';
+import { useOpprettBehandlingContext } from './OpprettBehandlingProvider';
 import OpprettBehandlingSkjema from './OpprettBehandlingSkjema';
+import Skjemasteg from '../Skjemasteg/Skjemasteg';
+import useFagsakApi from '../useFagsakApi';
+import { Behandlingstype } from '../../../typer/fagsak';
 
 const OpprettBehandling: React.FunctionComponent = () => {
-    const fagsakDispatcher = useFagsakDispatch();
-
-    const history = useHistory();
     const context = useOpprettBehandlingContext();
-    const dispatch = useOpprettBehandlingDispatch();
+
     const [visFeilmeldinger, settVisFeilmeldinger] = React.useState(false);
     const [opprettelseFeilmelding, settOpprettelseFeilmelding] = React.useState('');
 
+    const { opprettBehandling, senderInn } = useFagsakApi(
+        settVisFeilmeldinger,
+        settOpprettelseFeilmelding
```

Comment:
Maybe we can handle the exceptions and error messages in Skjemastag, so we don’t need to define these states for every page and the parameters to useFagsakApi can be simplified

Reply:
The problem is that then we need to define props for the children and this is problematic. We could solve this with providers, but I dont think we need this complexity.